infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactory.java

```diff
@@ -223,51 +223,36 @@ private ConfigMap newConfigMap(String configMapName, Collection<PluginFQN> plugi
     }
   }
 
+  private Volume newConfigMapVolume(String configMapName, String configMapVolume) {
+    return new VolumeBuilder()
+        .withName(configMapVolume)
+        .withNewConfigMap()
+        .withName(configMapName)
+        .endConfigMap()
+        .build();
+  }
+
   private EnvVar asEnvVar(Pair<String, String> envVar) {
     return new EnvVarBuilder().withName(envVar.first).withValue(envVar.second).build();
   }
 
-  private BrokerConfig createBrokerConfig(
-      RuntimeIdentity runtimeId,
-      @Nullable Collection<PluginFQN> pluginFQNs,
-      List<EnvVar> envVars,
-      String image,
-      Pod pod)
-      throws InternalInfrastructureException {
-
-    BrokerConfig brokerConfig = new BrokerConfig();
-    String configMapVolume = null;
-    if (pluginFQNs != null) {
-      brokerConfig.configMapName = generateUniqueName(CONFIG_MAP_NAME_SUFFIX);
-      brokerConfig.configMapVolume = generateUniqueName(BROKER_VOLUME);
-      brokerConfig.configMap = newConfigMap(brokerConfig.configMapName, pluginFQNs);
-      configMapVolume = brokerConfig.configMapVolume;
-    }
-    brokerConfig.container = newContainer(runtimeId, envVars, image, configMapVolume);
-    brokerConfig.machineName = Names.machineName(pod.getMetadata(), brokerConfig.container);
-    brokerConfig.machineConfig = new InternalMachineConfig();
-    brokerConfig
-        .machineConfig
-        .getVolumes()
-        .put(PLUGINS_VOLUME_NAME, new VolumeImpl().withPath("/plugins"));
-
-    return brokerConfig;
+  private String generateUniqueName(String suffix) {
+    return NameGenerator.generate(suffix, 6);
   }
 
-  private static class BrokerConfig {
-
-    String configMapName;
-    ConfigMap configMap;
-    Container container;
-    String machineName;
-    InternalMachineConfig machineConfig;
-    String configMapVolume;
+  /**
+   * Generate a container name from an image reference. Since full image references can be over 63
+   * characters, we need to strip registry and organization from the image reference to limit name
+   * length.
+   */
+  @VisibleForTesting
+  protected String generateContainerNameFromImageRef(String image) {
+    return image.toLowerCase().replaceAll("[^/]*/", "").replaceAll("[^\\d\\w-]", "-");
   }
 
   public static class BrokersConfigs {
```

Comment:
Such a minor comment:
The time we had multiple Brokers(theia, che, vscode) went passed, maybe it's a time to rename it to BrokerConfig.

P.S. If you mark it as resolved without response or action item - it's OK :smile: